sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/cryptography/CryptographyClientTest.java

```diff
@@ -32,35 +36,34 @@
     @Override
     protected void beforeTest() {
         beforeTestSetup();
-        if (interceptorManager.isPlaybackMode()) {
-            client = clientSetup(pipeline -> {
-                this.pipeline = pipeline;
-                return new KeyClientBuilder()
-                    .pipeline(pipeline)
-                    .vaultUrl(getEndpoint())
-                    .buildClient();
-            });
-        } else {
-            client = clientSetup(pipeline -> {
-                this.pipeline = pipeline;
-                return new KeyClientBuilder()
-                    .pipeline(pipeline)
-                    .vaultUrl(getEndpoint())
-                    .buildClient();
-            });
-        }
     }
 
-    @Test
-    public void encryptDecryptRsa() throws Exception {
+    private void initializeKeyClient(HttpClient httpClient) {
+        pipeline = getHttpPipeline(httpClient, KeyServiceVersion.getLatest());
+        client = new KeyClientBuilder()
+            .pipeline(pipeline)
+            .vaultUrl(getEndpoint())
+            .buildClient();
+    }
+
+    private CryptographyClient initializeCryptographyClient(String keyId, HttpClient httpClient, CryptographyServiceVersion serviceVersion) {
+        pipeline = getHttpPipeline(httpClient, serviceVersion);
```

Comment:
Shouldn't the same pipeline be used for both KeyClient and CryptographyClient?

Reply:
Not exactly, `initializeKeyClient` is using the latest KeyServiceVersion, as the main purpose of the test is to test over CryptographyServiceClient, which is using CryptographyServiceVersion. There are some other tests for KeyClient

Reply:
Before this change, the same pipeline was used for both.

Reply:
Talked offline. This does not affect the logic for tests. Besides of that, These two do not use exactly same pipelines. UserAgentPolicy are not the same.